src/Symfony/Component/Console/Helper/Table.php

```diff
@@ -205,24 +205,26 @@ public function setRow($column, array $row)
     public function render()
     {
         $this->calculateNumberOfColumns();
-        $this->rows = $this->buildTableRows($this->rows);
-        $this->headers = $this->buildTableRows($this->headers);
+        $rows = $this->buildTableRows($this->rows);
+        $headers = $this->buildTableRows($this->headers);
+
+        $this->calculateColumnsWidth(array_merge($headers, $rows));
```

Comment:
Instead of having this method change the state of the object, shouldn't we make it return $columnWidths? To me, the render() method would be better not mutating $this, if that makes sense.